### PR TITLE
HCK-7804: [RE] Snowflake dynamic tables

### PR DESCRIPTION
**Pull request by Nightlngale**
<table>
<td>
  <a href="https://hackolade.atlassian.net/browse/HCK-7804" title="HCK-7804" target="_blank">Sub-taskHCK-7804</a>  [RE] Snowflake dynamic tables
  </td></table>
  <br />
 

## Content

RE of dynamic tables

### CODE REVIEW

File: reverse_engineering/helpers/snowflakeHelper.js [hunk 1]
```diff
@@ -1136,6 +1149,120 @@ const getMaterializedViewData = async fullName => {
...
+			selectStatement,
+			externalVolume,
+			catalog,
+			initialize,
+			baseLocation,
+			DATA_RETENTION_TIME_IN_DAYS,
+			MAX_DATA_EXTENSION_TIME_IN_DAYS,
+		};
+	} catch (err) {
+		return {};
```
Comment (bigorn0) [inline reverse_engineering/helpers/snowflakeHelper.js:1262]: @Nightlngale please make sur to log errors and make them bubble if required.  Just silenting any error doesn't help on the long run :)

**Comment by bigorn0** [comment]
@Nightlngale updating deps (except in case of major) should part of usual dev maintenance.  Snowflake SDK got a few updates (for instances with addons like `Released Snowflake’s official d.ts type declaration file to support TypeScript users.` since the version we use https://github.com/hackolade/Snowflake/blob/ea98af7d7b8193fbf21017749c1e2c702ed51b73/package.json#L57 -> https://docs.snowflake.com/en/release-notes/clients-drivers/nodejs-2024

File: reverse_engineering/helpers/snowflakeHelper.js [hunk 2]
```diff
@@ -1136,6 +1149,120 @@ const getMaterializedViewData = async fullName => {
 	}
 };
 
+function getOptionValue(query, optionName) {
```
Comment (chulanovskyi-bs) [inline reverse_engineering/helpers/snowflakeHelper.js:1152]: please follow the "property bag" approach for new functions

File: reverse_engineering/helpers/snowflakeHelper.js [hunk 3]
```diff
@@ -1136,6 +1149,120 @@ const getMaterializedViewData = async fullName => {
...
+		const targetLag = getTargetLag(_.get(data, 'target_lag', ''));
+		const externalVolume = getOptionValue(text, 'EXTERNAL_VOLUME');
+		const initialize = getOptionValue(text, 'INITIALIZE').toLowerCase();
+		const catalog = getOptionValue(text, 'CATALOG');
+		const baseLocation = getOptionValue(text, 'BASE_LOCATION');
+		const DATA_RETENTION_TIME_IN_DAYS = getOptionValue(text, 'DATA_RETENTION_TIME_IN_DAYS');
+		const MAX_DATA_EXTENSION_TIME_IN_DAYS = getOptionValue(text, 'MAX_DATA_EXTENSION_TIME_IN_DAYS');
+
+		const selectStatement = text
+			.slice(text.indexOf('AS\n') + 4, -1)
```
Comment (chulanovskyi-bs) [inline reverse_engineering/helpers/snowflakeHelper.js:1242]: please get rid of magic numbers

**Review by VitaliiBedletskyi — APPROVED** [review]

**Pull request merged**